cannon/mipsevm/mips.go

```diff
@@ -246,6 +250,9 @@ func (m *InstrumentedState) handleHiLo(fun uint32, rs uint32, rt uint32, storeRe
 }
 
 func (m *InstrumentedState) handleJump(linkReg uint32, dest uint32) error {
+	if m.state.NextPC != m.state.PC+4 {
+		panic("jump in delay slot")
```

Comment:
why not return an error? Do we ever return a non-nil error?

Reply:
It's not ideal, but we panic instead, to isolate effects on the VM state. An invalid step may mess up the VM state.
We could save the pre-state and roll back when there's an invalid step, but doing so right now would be inefficient and double memory usage. Efficient VM snapshotting could be implemented but it's not a priority right now.